feat: trackID for Dialog Popover ActionList Dialog Footer  OK-37905

limichange opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added tracking and logging capabilities to Dialog, Popover, ActionList, and Dialog Footer components via a new optional `trackID` property.
  - UI events such as open, close, confirm, cancel, and toggle are now logged for analytics and monitoring.
  - Demo and gallery components updated to include unique `trackID` values showcasing tracking in action.
  - Dialog methods wrapped to automatically inject a fixed `trackID` for consistent event logging.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

limichange committed:
feat: add dialog tracking functionality and logger integration

- Introduced trackId property in IDialogFooterProps and IBasicDialogProps for enhanced tracking capabilities.
- Added DialogScene class to handle dialog-related logging with open, close, confirm, and cancel methods.
- Updated UIScope to include dialog scene for better UI tracking.

limichange committed:
feat: enhance dialog tracking with logger integration

- Added trackId support in Dialog and Footer components for improved tracking.
- Integrated logging for dialog open, confirm, and cancel actions using defaultLogger.
- Updated dependencies to ensure proper tracking functionality across dialog interactions.

limichange committed:
feat: enhance dialog close tracking with logger integration

- Added logging for dialog close actions using defaultLogger when trackId is present.
- Updated dependencies to ensure proper tracking functionality during dialog interactions.

limichange committed:
feat: enhance dialog close functionality with flag support

- Updated dialog close methods in Footer and Dialog components to include flags for confirm and cancel actions.
- Added trackId support in the Dialog story for improved tracking during testing.

limichange committed:
feat: add trackId support to Dialog and Content components

- Introduced trackId property in IDialogContentProps for enhanced tracking capabilities.
- Updated DialogFrame to pass trackId to Content and Footer components for improved tracking during dialog interactions.

limichange committed:
refactor: remove trackId from DialogFrame and enhance Dialog wrapper

- Removed trackId prop from DialogFrame's Content component.
- Created a wrapper for BaseDialog to automatically inject trackId into show, confirm, and cancel methods for consistent tracking.

limichange committed:
refactor: simplify dialog method return values

- Removed unnecessary object wrapping in open, close, confirm, and cancel methods of DialogScene, returning params directly for cleaner code.

limichange committed:
refactor: standardize trackId naming in Dialog components

- Updated all instances of trackId to trackID across Dialog, Footer, and related types for consistency.
- Ensured that tracking functionality remains intact with the new naming convention.

limichange committed:
refactor: enforce trackId consistency in Dialog components

- Updated trackID to trackId in DialogFrame and related methods for uniformity.
- Modified IDialogParams to require trackId, ensuring all dialog components adhere to the new naming convention.
- Enhanced Footer component to conditionally set trackID for confirm button.

limichange committed:
refactor: add trackID prop to BasicDialogHeader for enhanced tracking

- Updated BasicDialogHeader to accept an optional trackID prop for improved tracking capabilities.
- Modified DialogFrame to pass trackID to DialogHeader, ensuring consistent tracking across dialog components.

limichange committed:
feat: add trackID support to Popover component for enhanced tracking

- Introduced trackID prop in IPopoverProps to enable tracking of popover interactions.
- Updated usePopoverValue hook to log open and close events using defaultLogger when trackID is provided.
- Enhanced Popover stories to demonstrate trackID functionality for both controlled and uncontrolled scenarios.
- Added PopoverScene class for structured logging of popover events.

limichange committed:
feat: add trackID support to ActionList component for enhanced tracking

- Introduced trackID prop in IActionListProps to enable tracking of ActionList interactions.
- Implemented logging for ActionList open and close events using defaultLogger when trackID is provided.
- Updated ActionList stories to demonstrate trackID functionality across various scenarios.
- Created ActionListScene class for structured logging of ActionList events.

limichange committed:
fix: enable event tracking in Analytics class for development and E2E environments

- Re-enabled the conditional check for development and E2E environments in the trackEvent method to prevent event logging during testing phases.

limichange committed:
Merge branch 'x' into OK-37905

originalix commented on packages/components/src/composite/Dialog/Footer.tsx:
```diff
@@ -168,6 +177,11 @@ export function Footer(props: IDialogFooterProps) {
         <XStack p="$5" pt="$0" gap="$2.5" {...footerProps}>
           {showCancelButton ? (
             <Button
+              trackID={
+                trackID
+                  ? `${trackID}-cancel-button`
+                  : cancelButtonProps?.trackID
+              }
```
如果这么加的话，点击 cancel 的打点顺序就是两次对吧，能优化成一次么

dialogCancel -> click(`${trackID}-cancel-button`)

- 只监听 Button 的 click
- 只执行 dialogCancel 这类事件，除非指定 `cancelButtonProps?.trackID` 才进行 2 次打点

这两种方式取一种会不会有啥问题

limichange replied:
> 只执行 dialogCancel 这类事件，除非指定 cancelButtonProps?.trackID 才进行 2 次打点

这个比较省事，而且情况不多，因为二者都是相关的。而且数据上 dialogCancel 比较直观。把事件名更新了。

limichange committed:
fix: correct trackId to trackID in Dialog component methods for consistency

- Updated show, confirm, and cancel methods in the Dialog component to use trackID instead of trackId, ensuring uniformity with the naming convention established in previous commits.

limichange committed:
refactor: standardize logging method names in UI components

- Updated logging method names in ActionList, Popover, and Dialog components to follow a consistent naming convention, changing methods to actionListOpen, actionListClose, popoverOpen, popoverClose, dialogOpen, dialogClose, dialogConfirm, and dialogCancel.
- Ensured that all relevant instances in the logger classes were updated accordingly for uniformity.

sidmorizon commented on packages/shared/src/logger/scopes/ui/scenes/actionList.ts:
```diff
@@ -0,0 +1,27 @@
+  /**
+   * Additional custom properties
+   */
+  [key: string]: any;
```
埋点的自定义属性要显式指定，避免业务方随意传值泄漏敏感信息

originalix replied:
是的，kv 可以先删除，目前只追踪 trackId 也没问题

limichange replied:
好的。

limichange committed:
refactor: remove additional custom properties from UI component interfaces

- Eliminated the optional custom properties index signature from IActionListParams, IButtonClickParams, IDialogParams, and IPopoverParams interfaces to enforce stricter type definitions and improve clarity in the UI component logging structure.

limichange committed:
Merge branch 'x' into OK-37905

limichange committed:
Merge branch 'x' into OK-37905

sidmorizon committed:
Merge branch 'x' into OK-37905
